iterative/resource_runner.go

```diff
@@ -288,6 +281,47 @@ func resourceRunnerCreate(ctx context.Context, d *schema.ResourceData, m interfa
 	return diags
 }
 
+type resourceLogsResult struct {
+	logEvents string
+	logError  error
+}
+
+func resourceLogsTimed(ctx context.Context, d *schema.ResourceData, m interface{}) (string, error) {
+	result := make(chan resourceLogsResult, 1)
+	go func() {
+		result <- resourceLogs(ctx, d, m)
+	}()
+	select {
+	case <-time.After(20 * time.Second):
+		return "", errors.New("Resource Logs Timed Out")
+	case result := <-result:
+		return result.logEvents, result.logError
+	}
+}
```

Comment:
I love the smell of leaked _goroutines_ in the morning

Reply:
Not sure what you mean

Reply:
I have followed what is expected for a go timeout function

Reply:
I mean that the _goroutine_ from lines 291-293 can hang forever and leak; i.e. keep running after the outer function returns.

Reply:
aha... are they going to survive Terraform execution? Also... Im following the ssh devs recommendations

Reply:
Please suggest an additional fix but... right now it works and the leak is not going to persist by any meaning

Reply:
On solution might be move the goroutine to the ssh Dial and the close the conn if timeout

Reply:
> _Please suggest an additional fix_

I'd consider something like Teleport's fix mentioned on https://github.com/iterative/terraform-provider-iterative/pull/607#issuecomment-1159793257 🤷🏼‍♂️

Reply:
> _are they going to survive Terraform execution?_

No, once the program exits, all the _goroutines_ also exit. 🙃

Reply:
Thats what I say... I do not see a mayor issue... but I can improve it